internal/repository/postgres/storage.go

```diff
@@ -162,12 +238,20 @@ func (st *PsqlStorage) GetAllByUser(
 }
 
 func (st *PsqlStorage) Ping(ctx context.Context) (bool, error) {
-	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 	defer cancel()
 	err := st.db.PingContext(ctx)
 	if err != nil {
+		cancel()
 		return false, err
 	}
 	return true, nil
 }
+
+func (st *PsqlStorage) Close(_ context.Context) error {
+
+	st.stop <- struct{}{}
```

Comment:
```suggestion
	close(st.stop)
```
Так вернее. Если где-то еще понадобится среагировать на остановку, то закрытие канала инициирует переход по ветке сразу во всех этих местах.

Однако, на данном этапе при наличии в RunDelete не сохраненных данных мы гарантированно словим панику, так как успеем закрыть дескриптор к БД **до** того, как попытаемся их записать. Это хороший заход на graceful shutdown, но ему не хватает ожидания завершения работы RunDelete(). Для реализации следует создать WaitGroup и пробросить ее в RunDelete(). Можно даже привязать к репозиторию, главное - сделать перед запуском RunDelete wg.Add(), а внутри самой функции defer wg.Done(). Ну а здесь поставить wg.Wait()

Reply:
Почти хорошо, но я предлагаю `close(st.stop)` сделать **вместо** `st.stop <- struct{}{}`. Все select, настроенные на чтение из данного канала будут гарантированно срабатывать на закрытый канал. В случае передачи в канал единственного значения, на него сработает только один select. В настоящий момент он и есть один, но никто не может гарантировать, что не появится второй.